doc/source/user/absolute_beginners.rst

```diff
@@ -0,0 +1,2451 @@
+  np.random.random(3,2)
+
+.. image:: images/np_ones_zeros_matrix.png
+
+Read more about initializing the values of an array with `ones here <https://docs.scipy.org/doc/numpy/reference/generated/numpy.ones.html>`_, `zeros here <https://docs.scipy.org/doc/numpy/reference/generated/numpy.zeros.html>`_, and `initializing empty arrays here <https://docs.scipy.org/doc/numpy/reference/generated/numpy.empty.html>`_.
+
+
+Generating random numbers
+-------------------------
+
+The use of random number generation is an important part of the configuration and evaluation of machine learning algorithms. Whether you need to randomly initialize weights in an artificial neural network, split data into random sets, or randomly shuffle your dataset, being able to generate random numbers (actually, repeatable pseudo-random numbers) is essential.
+
+You have a number of options when using NumPy for random number generation. Random Generator is NumPy's replacement for RandomState. The main difference between them is that Generator relies on an additional BitGenerator to manage state and generate the random bits, which are transformed into random values.
+
+With Generator.integers, you can generate random integers from low (remember that this is inclusive with NumPy) to high (exclusive). You can set ``endopoint=True`` to make the high number inclusive. 
+
+You can generate a 2 x 4 array of random integers between 0 and 4 with:
+
+**Input:**
+
+::
+
+  rng.integers(5, size=(2, 4))
+
+**Output:**
+
+::
+
+  array([[4, 0, 2, 1],
+       [3, 2, 2, 0]])
+
+`Read more about Random Generator here <https://docs.scipy.org/doc/numpy/reference/random/generator.html>`_.
+
+
+How to get unique items and counts
+------------------------------------------
+
+::
+
+  np.unique()
+
+-----
+
+You can find the unique elements in an array easily with ``np.unique``. 
+
+For example, if you start with this array:
+
+**Input:**
+
+::
+
+  a = np.array([11, 11, 12, 13, 14, 15, 16, 17, 12, 13, 11, 14, 18, 19, 20])
+
+you can use ``np.unique``
+
+**Input:**
+
+::
+
+  unique_values = np.unique(a)
+  print(unique_values)
+
+**Output:**
+
+::
+
+  [11 12 13 14 15 16 17 18 19 20]
+
+To get the indices of unique values in a NumPy array (an array of first index positions of unique values in the array), just pass the ``return_index`` argument in ``np.unique()`` as well as your array.
+
+**Input:**
+
+::
+
+  indices_list = np.unique(a, return_index=True)
+  print(indices_list)
+
+**Output:**
+
+::
+
+  [ 0  2  3  4  5  6  7 12 13 14]
+
+You can pass the ``return_counts`` argument in ``np.unique()`` along with your array to get the frequency count of unique values in a NumPy array.
+
+**Input:**
+
+::
+
+  unique_values, occurrence_count = np.unique(a, return_counts=True)
+  print(occurrence_count)
+
+**Output:**
+
+::
+
+  [3 2 2 2 1 1 1 1 1 1]
+
+This also works with 2D arrays. If you start with this array:
+
+::
+
+  a2D = np.array([[1, 2, 3, 4] ,[5, 6, 7, 8] , [9, 10, 11, 12],  [1, 2, 3, 4]])
```

Comment:
some misplaces commas, and name would preferably be `a_2d`

Reply:
Got it!